### ISSUE TITLE
[P3][M1] Sema comment says 'Full type checking is TODO' — no longer accurate

### ISSUE THREAD

**Issue by ChaseWNorton** [anthropics/claudes-c-compiler#25]
## Problem

`src/frontend/sema/analysis.rs` lines 15-16 say:

```
//! This pass does NOT reject programs with type errors (yet); it collects
//! information for the lowerer. Full type checking is TODO.
```

This was accurate when the file was first written, but after numerous sema improvements (undeclared variable errors, storage class validation, const qualification, linkage checking, implicit conversions, pointer type warnings, integer overflow diagnostics, switch type checking, incomplete struct detection, etc.), it is misleading.

The sema pass now performs substantial type checking and **does** reject programs with errors. The comment should reflect current reality.

## Expected behavior

The module comment should accurately describe what the sema pass does today: it performs type checking, emits errors and warnings, and rejects invalid programs.

## Suggested approach

Update the module doc comment at lines 1-16 of `src/frontend/sema/analysis.rs` to accurately describe what the pass does. Something like:

```rust
//! Semantic analysis pass.
//!
//! Walks the AST to:
//! - Build a scoped symbol table of declarations
//! - Track function signatures for call validation
//! - Resolve typedef names and typeof(expr) via ExprTypeChecker
//! - Perform type checking: diagnose type errors, invalid conversions,
//!   storage class conflicts, linkage violations, and other constraints
//! - Emit warnings for suspicious patterns (-Wall/-Wextra categories)
//! - Collect information needed by the IR lowering phase
//! - Map __builtin_* identifiers to their libc equivalents
//!
//! Programs with errors are rejected (analyze() returns Err with error count).
//! The diagnostic engine handles warning filtering (-Wno-*) and -Werror promotion.
```

## Files to modify

- `src/frontend/sema/analysis.rs` — Update the module doc comment (lines 1-16)

## Tests

No new tests needed — this is a documentation fix. Verify with:
- `cargo build --release` (clean build)
- `cargo test --lib` (all tests pass)

## Acceptance criteria

- [ ] Module comment accurately describes current sema capabilities
- [ ] The word "TODO" is removed from the comment
- [ ] Clean build: `cargo build --release`
- [ ] All existing tests pass: `cargo test --lib`

Part of [MILESTONE] M1: Core Diagnostic Coverage (#40)
